apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_connector.erl

```diff
@@ -623,45 +615,41 @@ on_add_channel(
             {error, already_exists};
         _ ->
             %% update HTTP channel
-            InsertTabletPathV1 = <<"rest/v1/insertTablet">>,
-            InsertTabletPathV2 = <<"rest/v2/insertTablet">>,
-
-            Path =
-                case Version of
-                    ?VSN_1_1_X -> InsertTabletPathV2;
-                    ?VSN_1_3_X -> InsertTabletPathV2;
-                    _ -> InsertTabletPathV1
-                end,
-
-            HTTPReq = #{
-                parameters => Parameter#{
-                    path => Path,
-                    method => <<"post">>
-                }
-            },
-
-            {ok, OldState} = emqx_bridge_http_connector:on_add_channel(
-                InstanceId, OldState0, ChannelId, HTTPReq
-            ),
-
-            %% update IoTDB channel
-            DeviceId = maps:get(device_id, Parameter, <<>>),
-            Channel = Parameter#{
-                device_id => emqx_placeholder:preproc_tmpl(DeviceId),
-                data := preproc_data_template(Data)
-            },
-            Channels2 = Channels#{ChannelId => Channel},
-            {ok, OldState#{channels := Channels2}}
+            case Version of
+                ?VSN_1_3_X ->
+                    Path = <<"rest/v2/insertRecords">>,
+                    HTTPReq = #{
+                        parameters => Parameter#{
+                            path => Path,
+                            method => <<"post">>
+                        }
+                    },
+
+                    {ok, OldState} = emqx_bridge_http_connector:on_add_channel(
+                        InstanceId, OldState0, ChannelId, HTTPReq
+                    ),
+
+                    %% update IoTDB channel
+                    DeviceId = maps:get(device_id, Parameter, <<>>),
+                    Channel = Parameter#{
+                        device_id => emqx_placeholder:preproc_tmpl(DeviceId),
+                        data := preproc_data_template(Data)
+                    },
+                    Channels2 = Channels#{ChannelId => Channel},
+                    {ok, OldState#{channels := Channels2}};
+                _ ->
+                    {error, <<"REST API only supports IoTDB 1.3.x and later">>}
+            end
     end;
 on_add_channel(
     _InstanceId,
     #{driver := thrift},
     _ChannelId,
     #{
-        resource_opts := #{query_mode := QueryMode, batch_size := BatchSize}
+        resource_opts := #{query_mode := async}
     }
-) when QueryMode =:= async; BatchSize > 1 ->
-    {error, <<"Thrift does not support async or batch mode">>};
+) ->
+    {error, <<"Thrift does not support async mode">>};
```

Comment:
Maybe this could be a static schema validation.  i.e., don't even call this callback, the schema should fail to validate.

Reply:
I think this is impossible. This parameter is from the action, but the `driver` is an attribute for the connector, so this can only be determined in runtime.